External page components/Public links preview.js

```diff
@@ -96,58 +98,83 @@ PublicLinkOverview.prototype = {
+                    });
+                });
             }
         }.bind(self));
     }
+}
+
+function compareResponse(first, second) {
+    var firstTitle = first["properties"]["RelativeUrl"];
+    var secondTitle = second["properties"]["RelativeUrl"];
+
+    return ('' + firstTitle).localeCompare(secondTitle, undefined, { numeric: true, sensitivity: 'base' });
+}
+
+function convertToHtml(publicLinkData) {
+    // retrieve the entity data of each public link and get the public link url to add a thumbnail to the overview
+    var publicLink = publicLinkData["public_link"];
+    if (publicLink && publicLink !== "") {
+
+        // the thumbnail is displayed using a transformation to avoid loading large images
+        // to ensure the public link thumbnail isn't cached, we add a random value to the image source
+        var thumbnail = "";
+        if (self.plo._renditions[publicLinkData["properties"]["Resource"]].content_type.startsWith("image")) {
+            // the thumbnail is only displayed for image type renditions (not for Documents for example, or other application type public links)
+            thumbnail = publicLink + "&t=thumbnail&r=" + Math.random();
+        }
+
+        var title = publicLinkData["properties"]["RelativeUrl"];
+        var rendition = self.plo._renditions[publicLinkData["properties"]["Resource"]].label;
+        var width = "";
+        var height = "";
+
+        var croppingType = null;
+        if (publicLinkData["properties"]["ConversionConfiguration"] && publicLinkData["properties"]["ConversionConfiguration"]["cropping_configuration"]) {
+            croppingType = publicLinkData["properties"]["ConversionConfiguration"]["cropping_configuration"]["cropping_type"];
+            width = publicLinkData["properties"]["ConversionConfiguration"]["width"];
+            height = publicLinkData["properties"]["ConversionConfiguration"]["height"];
+
+            // depending on the way of cropping, the width and height are stored in different objects
+            if (!width || width == null) {
+                width = publicLinkData["properties"]["ConversionConfiguration"]["cropping_configuration"]["width"];
+                height = publicLinkData["properties"]["ConversionConfiguration"]["cropping_configuration"]["height"];
+            }
+        }
+
+        if (croppingType === "Entropy") {
+            croppingType = "Smart crop";
+        } else if (croppingType === "Custom") {
+            croppingType = "Custom crop";
+        } else if (croppingType === "CentralFocalPoint") {
+            croppingType = "Crop to center";
+        } else {
+            croppingType = "Uncropped";
+        }
+
+        var context = { title: title, href: publicLink, preview: thumbnail, rendition: rendition, width: width, height: height, croppingType: croppingType };
+        var html = listItemTemplate(context);
+        document.getElementById("publicLinkList").innerHTML += html;
```

Comment:
You can leave this over here, so it'll add the HTML to the publicLinkList element, or move it over to line 120, converting this method to a method returning the html only instead of adding it to the page. Either way would work, but I feel it doesn't add anything to the code transforming this in a return method. For clarity, I would keep it here.